src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -527,8 +531,12 @@ private BoundExpression GetExtensionMemberAccess(SyntaxNode syntax, BoundExpress
                 case NamedTypeSymbol namedTypeSymbol:
                     bool wasError = false;
 
-                    return BindTypeMemberOfType(receiver, namedTypeSymbol.Name, namedTypeSymbol,
-                        syntax, right: syntax, diagnostics, ref wasError);
+                    if (!typeArgumentsOpt.IsDefault)
+                    {
+                        namedTypeSymbol = ConstructNamedTypeUnlessTypeArgumentOmitted(syntax, namedTypeSymbol, typeArgumentsSyntax, typeArgumentsOpt, diagnostics);
```

Comment:
>namedTypeSymbol [](http://example.com/codeflow?start=24&length=15)

When and where do we check constraints for this type?

Reply:
The constraints get checked a couple of layers into `ConstructNamedTypeUnlessTypeArgumentOmitted`:
`ConstructNamedTypeUnlessTypeArgumentOmitted` -> `ConstructNamedType` -> `CheckConstraintsForNamedType`.
Example test: `InferredVariable_TypeReceiver_GenericType_BrokenConstraint`